openidm-workflow-activiti/src/main/java/org/forgerock/openidm/workflow/activiti/impl/OpenIDMResolverFactory.java

```diff
@@ -89,18 +91,22 @@ public Resolver createResolver(VariableScope variableScope) {
 
         try {
             if (variableScope instanceof ExecutionEntity) {
-                ActivityBehavior activityBehavior = ((ExecutionEntity) variableScope).getActivity().getActivityBehavior();
-                //Called from ScriptTask
-                if (activityBehavior instanceof ScriptTaskActivityBehavior) {
-                    behaviour = (ScriptTaskActivityBehavior) activityBehavior;
-                    Class<?> cls = Class.forName("org.activiti.engine.impl.bpmn.behavior.ScriptTaskActivityBehavior");
+                ExecutionEntity execution = (ExecutionEntity) variableScope;
+                // Called from ScriptTask
+                if (execution.getActivity() != null && execution.getActivity().getActivityBehavior() instanceof ScriptTaskActivityBehavior) {
```

Comment:
can you put `execution.getActivity()` into a local var so we don't fetch it so many times in this block?

Reply:
There are few other issues with the code, not connected to this PR as well... like why the heck there is `Class.forName` if the class is also being imported :).